conan/tools/env/virtualbuildenv.py

```diff
@@ -12,21 +12,27 @@ def __init__(self, conanfile):
         self._conanfile = conanfile
         self._conanfile.virtualbuildenv = False
         self.basename = "conanbuildenv"
-        # TODO: Make this use the settings_build
-        self.configuration = conanfile.settings.get_safe("build_type")
-        if self.configuration:
-            self.configuration = self.configuration.lower()
-        self.arch = conanfile.settings.get_safe("arch")
-        if self.arch:
-            self.arch = self.arch.lower()
+        self.configuration = None
+        self.arch = None
 
     @property
     def _filename(self):
+        if not self.configuration:
```

Comment:
Let's keep this change even if no longer necessary - feel more correct to me this way